onyo/lib/inventory.py

```diff
@@ -304,15 +305,17 @@ def add_asset(self, asset: dict | UserDict) -> list[InventoryOperation]:
         self.raise_required_key_empty_value(asset)
         name = self.generate_asset_name(asset)
 
-        if asset.get('is_asset_directory', False):
-            # 'path' needs to be given, if this is about an already existing dir.
-            # Otherwise, a 'directory' to create the asset in is expected as with
+        if asset.get('onyo.is.directory', False):
```

Comment:
`asset['onyo.is.directory']` should always return `True` or `False`, so the `get` doesn't provide any value here, except possibly obscuring some bugs.

Something to keep in mind when the later PRs come and start using these new pseudokeys internally.

Reply:
Nope.  That's the ultimate values for an existing thing in the inventory, yes. But when a new asset is created as an `Item`/dict and passed to `Inventory.add_asset` this property may not be specified (at least as of now) and can't be derived from the state of things on disk, because it doesn't exist yet. It's this very function that says "well, if it wasn't specified that counts as `False`".
We may want to change that and require this to be set by `onyo_new` instead, before it's passed here. Not opposing that, it's just not (yet) what's happening.

Reply:
Ahh. Thanks for the explanation.